### ISSUE TITLE
autodoc: class __init__ override not removed from docstring

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1138]
If `autodoc_docstring_signature = True` and the user provides an override for a class `__init__` method signature the signature is properly assigned, but the string specifying the override remains part of the docstring, unlike with other methods.

For example, the following code:

```
#!python

class MyClass(object):
    def __init__(self, *args, **kwargs):
        """
        __init__(arg1, arg2, kwarg1=False, kwarg2=None)

        Docstring with signature override.
        """
        pass

    def my_method(self, *args, **kwargs):
        """
        my_method(arg1, arg2, kwarg1=False, kwarg2=None)

        Another docstring with signature override.
        """
        pass
```

This will produce something like:

```
class MyClass(arg1, arg2, kwarg1=False, kwarg2=None)
    __init__(arg1, arg2, kwarg1=False, kwarg2=None)

    Docstring with signature override.

    my_method(arg1, arg2, kwarg1=False, kwarg2=None)
        Another docstring with signature override.
```

As you can see the `__init__` override string remains part of the docstring, but the `my_method` override string is automatically removed as it should be.

This is with 1.2b1.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1138
- Originally reported by: Andrew Dawson
- Originally created at: 2013-04-08T16:19:09.550

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-11-17 09:31:05+00:00_

I reproduced this issue with:

conf.py:

```
#!python
autodoc_docstring_signature = True
autoclass_content = 'init'  #or 'both'
```

index.rst:

```
#!rst
.. autoclass:: mod.MyClass
```

mod.py:

```
#!python
class MyClass(object):

    def __init__(self, *args, **kwargs):
        """
        __init__(arg1, arg2, kwarg1=False, kwarg2=None)

        Docstring with signature override.
        """
        pass
```

This will produce something like:

```
#!python
class mod.MyClass(arg1, arg2, kwarg1=False, kwarg2=None)
    __init__(arg1, arg2, kwarg1=False, kwarg2=None)

    Docstring with signature override.
```

The `__init__` line should be eliminated.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-11-17 11:01:20+00:00_

The `__init__` line would be deleted when `autodoc_docstring_signature = False` ?

A. Should keep `__init__` line (Sphinx-1.x compatible):

```
#!python
class mod.MyClass(*args, **kwargs)
    __init__(arg1, arg2, kwarg1=False, kwarg2=None)

    Docstring with signature override.
```

B. Should remove `__init__` line (introduce incompatibility):

```
#!python
class mod.MyClass(*args, **kwargs)
    Docstring with signature override.
```

The attached `1138-remove-docstring-signature-line.patch` is A.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-12-08 05:03:41+00:00_

[Andrew Dawson](https://bitbucket.org/ajdawson) (or somebody) what do you think? 

If there is no opinion, this change will be carried over to the next release.

**Comment by shimizukawa** [comment]
_From Andrew Dawson on 2013-12-09 09:55:57+00:00_

This fixes my original bug. I think it produces nice consistent documentation so I am in favour of it. Thanks for the effort [Takayuki Shimizukawa](https://bitbucket.org/shimizukawa).

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-12-09 10:56:43+00:00_

Fix: autodoc class **init** override not removed from docstring. Closes #1138

→ <<cset 1d5c7d1f2014fb3f78b13218f2699db33555da4b>>
